Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -840,23 +839,20 @@ extension TaskHandler: ChannelDuplexHandler {
             self.writeBody(request: request, context: context)
         }.flatMap {
             context.eventLoop.assertInEventLoop()
+
             if case .endOrError = self.state {
```

Comment:
could we replace this `if case` and the fallthrough by one full `switch self.state`?

It's unclear to me why we can move from almost any state to `.sent` and why all those transitions are legal.

Reply:
For example a transition from `.sent` to `.sent` seems odd, right? Or from `.inactive` to `.sent` seems wrong too.

Reply:
done